code/edgemedia/src/test/java/com/adobe/marketing/mobile/edge/media/internal/MediaXDMEventHelperTests.kt

```diff
@@ -0,0 +1,216 @@
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMQoeDataDetails
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMSessionDetails
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMStreamType
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class MediaXDMEventHelperTests {
+    private val mediaStandardMetadata = mutableMapOf(
+        MediaConstants.VideoMetadataKeys.AD_LOAD to "adLoad",
+        MediaConstants.VideoMetadataKeys.ASSET_ID to "assetID",
+        MediaConstants.VideoMetadataKeys.AUTHORIZED to "authorized",
+        MediaConstants.VideoMetadataKeys.DAY_PART to "dayPart",
+        MediaConstants.VideoMetadataKeys.EPISODE to "episode",
+        MediaConstants.VideoMetadataKeys.FEED to "feed",
+        MediaConstants.VideoMetadataKeys.FIRST_AIR_DATE to "firstAirDate",
+        MediaConstants.VideoMetadataKeys.FIRST_DIGITAL_DATE to "firstDigitalDate",
+        MediaConstants.VideoMetadataKeys.GENRE to "genre",
+        MediaConstants.VideoMetadataKeys.MVPD to "mvpd",
+        MediaConstants.VideoMetadataKeys.NETWORK to "network",
+        MediaConstants.VideoMetadataKeys.ORIGINATOR to "originator",
+        MediaConstants.VideoMetadataKeys.RATING to "rating",
+        MediaConstants.VideoMetadataKeys.SEASON to "season",
+        MediaConstants.VideoMetadataKeys.SHOW to "show",
+        MediaConstants.VideoMetadataKeys.SHOW_TYPE to "showType",
+        MediaConstants.VideoMetadataKeys.STREAM_FORMAT to "streamFormat",
+
+        MediaConstants.AudioMetadataKeys.ALBUM to "album",
+        MediaConstants.AudioMetadataKeys.ARTIST to "artist",
+        MediaConstants.AudioMetadataKeys.AUTHOR to "author",
+        MediaConstants.AudioMetadataKeys.LABEL to "label",
+        MediaConstants.AudioMetadataKeys.PUBLISHER to "publisher",
+        MediaConstants.AudioMetadataKeys.STATION to "station"
+    )
+    private val adStandardMetadata = mutableMapOf(
+        MediaConstants.AdMetadataKeys.ADVERTISER to "advertiser",
+        MediaConstants.AdMetadataKeys.CAMPAIGN_ID to "campaignID",
+        MediaConstants.AdMetadataKeys.CREATIVE_ID to "creativeID",
+        MediaConstants.AdMetadataKeys.CREATIVE_URL to "creativeURL",
+        MediaConstants.AdMetadataKeys.PLACEMENT_ID to "placementID",
+        MediaConstants.AdMetadataKeys.SITE_ID to "siteID"
+    )
+
+    private val mediaInfo = MediaInfo.create("id", "name", "vod", MediaType.Video, 10.0)
+    private val adBreakInfo = AdBreakInfo.create("name", 1, 2.3)
+    private val adInfo = AdInfo.create("id", "name", 1, 10.0)
+    private val chapterInfo = ChapterInfo.create("name", 1, 2.3, 10.0)
+    private val qoeInfo = QoEInfo.create(1.1, 2.2, 3.3, 4.4)
+    private val muteStateInfo = StateInfo.create(MediaConstants.PlayerState.MUTE)
+    private val testStateInfo = StateInfo.create("testStateName")
+
+    val metadata = mutableMapOf("key1" to "value1", "key2" to "value2")
```

Comment:
For this custom metadata test data, place the values in a random order so the tests also verify the metadata is sorted.